torch/quantization/fx/utils.py

```diff
@@ -346,31 +346,34 @@ def create_qparam_nodes(quantizer: QuantizerCls, node_name: str, scale: Any, zer
     return (scale_node, zero_point_node)
 
 
-def all_node_args_have_no_tensors(node: Node, modules: Dict[str, torch.nn.Module]) -> bool:
+def all_node_args_have_no_tensors(node: Node, modules: Dict[str, torch.nn.Module], cache: Optional[Dict[Node, bool]] = None) -> bool:
```

Comment:
optional: we could also do something like

```
def all_node_args_have_no_tensors(node, modules, cache):
  if node in cache:
    return cache[node]
  result = _all_node_args_have_no_tensors_impl(node, modules)
  cache[node] = result
  return result
```

either is good

Reply:
thanks, this looks cleaner and less error prone

Reply:
actually, I think we still need to modify the original function to include cache..